### ISSUE TITLE
Object reference not set to an instance of an object when using the cancellation token

### ISSUE THREAD

**Issue by pavlexander** [CommunityToolkit/dotnet#579]
### Describe the bug

I am trying to use the cancellation token as described in documentation here [RelayCommand](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/relaycommand#asynchronous-commands)

```
[RelayCommand]
private async Task GreetUserAsync(CancellationToken token)
{
    try
    {
        User user = await userService.GetCurrentUserAsync(token);

        Console.WriteLine($"Hello {user.Name}!");
    }
    catch (OperationCanceledException)
    {
    }
}
```

However, whenever I add the cancellation token to the method - the program starts throwing the `Object reference not set to an instance of an object` exception when invoking the command.

### Steps to reproduce

1. using .Net 6.0 Avalonia MVVM app on Windows 10 x64, `CommunityToolkit.Mvvm v.8.1.0`, rest of the information along with full repro code you will find in a repo https://github.com/pavlexander/avaloniaExceptionTest1

2. Create fresh `Avalogia` app, add the community toolkit `nuget` package
3. Add a new dummy window
4. add 2 buttons to the main widnow. 

```xml
  <StackPanel>
    <Button Command="{Binding OpenNewWindowWithToken}">Open new window (with CancellationToken)</Button>
    <Button Command="{Binding OpenNewWindow}">Open new window</Button>
  </StackPanel>
```

5. create 2 commands in the main view model, one with cancellation token, one without

```cs
    [RelayCommand]
    public async Task OpenNewWindowWithToken(CancellationToken token)
    {
        var newWindow = new Window1();
        newWindow.Show();
    }

    [RelayCommand]
    public async Task OpenNewWindow()
    {
        var newWindow = new Window1();
        newWindow.Show();
    }
```

6. Launch the app. First click on the button that uses command without cancellation token. Works Ok.
7. Now click on the button that has cancellation token as an input parameter. Observe exception.

Stacktrace: 

>    at Avalonia.Controls.Button.OnClick()\r\n   at Avalonia.Controls.Button.OnPointerReleased(PointerReleasedEventArgs e)\r\n   at System.Reactive.Subjects.Subject`1.OnNext(T value)\r\n   at Avalonia.Interactivity.EventRoute.RaiseEventImpl(RoutedEventArgs e)\r\n   at Avalonia.Interactivity.EventRoute.RaiseEvent(IInteractive source, RoutedEventArgs e)\r\n   at Avalonia.Interactivity.Interactive.RaiseEvent(RoutedEventArgs e)\r\n   at Avalonia.Input.MouseDevice.MouseUp(IMouseDevice device, UInt64 timestamp, IInputRoot root, Point p, PointerPointProperties props, KeyModifiers inputModifiers)\r\n   at Avalonia.Input.MouseDevice.ProcessRawEvent(RawPointerEventArgs e)\r\n   at Avalonia.Input.InputManager.ProcessInput(RawInputEventArgs e)\r\n   at Avalonia.Win32.WindowImpl.AppWndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam)\r\n   at Avalonia.Win32.WindowImpl.WndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam)\r\n   at Avalonia.Win32.Interop.UnmanagedMethods.DispatchMessage(MSG& lpmsg)\r\n   at Avalonia.Win32.Win32Platform.RunLoop(CancellationToken cancellationToken)\r\n   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken)\r\n   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args)\r\n   at AvaloniaApplication2.Program.Main(String[] args) in C:\\repos\\avaloniaExceptionTest1\\AvaloniaApplication2\\Program.cs:line 14

### Expected behavior

I expect the method to work, no exceptions to be thrown, as documented.

### Screenshots

Here is the exception screenshot:

![image](https://user-images.githubusercontent.com/20644772/213916129-fb793bda-bc69-4482-9cbf-347d16b2d840.png)

### IDE and version

VS 2022

### IDE version

Microsoft Visual Studio Community 2022 (64-bit) - Current Version 17.4.3

### Nuget packages

- [ ] CommunityToolkit.Common
- [ ] CommunityToolkit.Diagnostics
- [ ] CommunityToolkit.HighPerformance
- [X] CommunityToolkit.Mvvm (aka MVVM Toolkit)

### Nuget package version(s)

CommunityToolkit.Mvvm v.8.1.0

### Additional context

I have created a repo that contains full code that you can check.. https://github.com/pavlexander/avaloniaExceptionTest1

### Help us help you

No, just wanted to report this

**Comment by Sergio0694** [comment]
I can repro this, but the exception seems to be somewhere in `Button.cs` in Avalonia, not caused by the MVVM Toolkit per se. Can you open a bug in Avalonia and link this issue there? This feels external, unless you can also provide a minimal repro of something not working as you think it should with just the MVVM Toolkit in a standalone project, without any Avalonia references.

Closing this for now, but feel free to reopen if anything changes 🙂
Thank you!

**Comment by Sergio0694** [comment]
Seeing this:

![](https://user-images.githubusercontent.com/10199417/213917795-59880dbb-ea27-4ff6-93c2-db410002d7a5.png)

**Comment by pavlexander** [comment]
@Sergio0694 Thank you for a quick response! Per your suggestion I have reported the issue to Avalonia (https://github.com/AvaloniaUI/Avalonia/issues/10048)

Regarding your comment - I can now also see the same exception as you are! I had to disable "Just my code" in VS debugger settings:

![image](https://user-images.githubusercontent.com/20644772/213924176-7c70de4f-f331-4f2d-a46f-ac274bc97574.png)

And also set the Exceptions config as:

![image](https://user-images.githubusercontent.com/20644772/213924203-40dd7209-20ec-4d2f-9df2-f3c5fa0b0ea3.png)

Now I see the same kind of message

![image](https://user-images.githubusercontent.com/20644772/213924227-0d6ce5a0-30ad-4a38-b492-4b2cc2c8ea06.png)

Let's wait and see what the Avalonia guru have to say on this matter.
